pkg/api/meta/meta.go

```diff
@@ -283,15 +245,15 @@ func (resourceAccessor) SetUID(obj runtime.Object, uid types.UID) error {
 }
 
 func (resourceAccessor) SelfLink(obj runtime.Object) (string, error) {
-	accessor, err := Accessor(obj)
+	accessor, err := ListAccessor(obj)
```

Comment:
All types that are Lists can have SelfLinks, but don't implement Accessor.  So the heart of this change is ensuring that we try to find the minimal expected interface (ListMeta) that can set / retrieve selfLinks.